### PR TITLE
Adding login Feature to client and added session middleware with route

### PR DESCRIPTION
**Pull request by 7-Dany**

### CODE REVIEW

File: Books/client/src/features/Login/components/SignIn.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  setLogin: React.Dispatch<React.SetStateAction<boolean>>
+}
+
+function SignIn({ setLogin }: SignInProps) {
+  const { formik, errorMsg, register } = useSignIn({ setLogin })
+  const { touched, handleSubmit, handleBlur, handleChange, errors, values } = formik
+  return (
+    <div className='login-form-container'>
+      <h2>Login</h2>
+      <p className='login-form__error-msg'>{errorMsg}</p>
```
Comment (htmlHxcker) [inline Books/client/src/features/Login/components/SignIn.tsx:14]: An error message here, how would that work? I thought error messages were supposed to display under the input field.

File: Books/client/src/features/Login/index.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { useState } from 'react'
+import SignIn from './components/SignIn'
+import SignUp from './components/SignUp'
+
+function Login() {
+  // To show sign in page if login is true and show sign up page if login is false
+  const [login, setLogin] = useState(true)
+  return (
+    <div>
+      {login ? <SignIn setLogin={setLogin} /> : <SignUp setLogin={setLogin} />}
```
Comment (htmlHxcker) [inline Books/client/src/features/Login/index.tsx:10]: Nitpick formatting issue but do you think you could put each condition on a separate line i.e
```js
{
login
 ? <SignIn setLogin={setLogin} /> 
: <SignUp setLogin={setLogin} />
}
```

**Review by htmlHxcker — APPROVED** [review]
Awesome work dude!

File: Books/client/src/features/Login/components/SignIn.tsx (hunk 1 above)
Comment (7-Dany) [reply Books/client/src/features/Login/components/SignIn.tsx:14]: That error msg from the server for example if the password is wrong or the user is not exist

File: Books/client/src/features/Login/index.tsx (hunk 2 above)
Comment (7-Dany) [reply Books/client/src/features/Login/index.tsx:10]: sure next time i will do that

**Pull request merged**